src/main/java/io/appform/jsonrules/expressions/equality/EqualsExpression.java

```diff
@@ -35,20 +35,46 @@
 @EqualsAndHashCode(callSuper = true)
 @ToString(callSuper = true)
 public class EqualsExpression extends JsonPathBasedExpression {
-    private Object value;
+	private Object value;
+	private boolean extractValueFromPath;
 
-    public EqualsExpression() {
-        super(ExpressionType.equals);
-    }
+	public EqualsExpression() {
+		super(ExpressionType.equals);
+	}
 
-    @Builder
-    public EqualsExpression(String path, Object value, PreOperation<?> preoperation) {
-        super(ExpressionType.equals, path, preoperation);
-        this.value = value;
-    }
+	@Builder
+	public EqualsExpression(String path, Object value, boolean extractValueFromPath, boolean defaultResult,
+			PreOperation<?> preoperation) {
+		super(ExpressionType.equals, path, defaultResult, preoperation);
+		this.value = value;
+		this.extractValueFromPath = extractValueFromPath;
+	}
 
-    @Override
-    protected boolean evaluate(ExpressionEvaluationContext context, String path, JsonNode evaluatedNode) {
-        return !ComparisonUtils.isNodeMissingOrNull(evaluatedNode) && ComparisonUtils.compare(evaluatedNode, value) == 0;
-    }
+	public EqualsExpression(String path, Object value, boolean extractValueFromPath, PreOperation<?> preoperation) {
+		this(path, value, extractValueFromPath, false, preoperation);
+	}
+
+	@Override
+	protected boolean evaluate(ExpressionEvaluationContext context, String path, JsonNode evaluatedNode) {
+		if (extractValueFromPath) {
+			JsonNode jsonNode = context.getNode().at(String.valueOf(value));
+			if (jsonNode.isNumber()) {
+				if (jsonNode.isIntegralNumber()) {
+					return !ComparisonUtils.isNodeMissingOrNull(evaluatedNode)
```

Comment:
isNodeMissingOrNull can be checked once globally before going into the decision tree?

Reply:
Done